### PR TITLE
add send-to-existing-session path to browser tab popovers

### PR DESCRIPTION
**Pull request by MusharrafSakran**
## Summary

Adds a "send to existing session" path in the Browser Tab pane's Draw / Flow-recording / Inspect popovers. Instead of always spawning a new session, captures (screenshot / flow / element selection) can now target an already-running or dormant Claude Code session.

- **Primary button** dynamically labels itself `Send to {active session name}` and one-click-sends to the smart default target (first running CLI session → first dormant → falls back to creating new).
- **Overflow `⋯` menu** opens a single inline popover listing all eligible CLI sessions (with status dots 🟢 running / ⚪ not started / 🔴 exited) plus `+ New session` and `+ New session with custom args…`.
- **Under the hood**: two delivery primitives — `injectPromptIntoRunningSession` (bracketed-paste into the PTY, mirroring the existing Windows Ctrl+V pattern in `terminal-utils.ts:55`) for running sessions; `setPendingPrompt` (existing `startup-arg` path) for dormant sessions. The `deliver()` helper tries inject first, falls back to pending.

Related: #76 — same "inject into active terminal" DNA, different surface (Agents panel).

## Test plan

- [x] `npm run build` passes
- [x] `npm test` passes (77 files / 1169 tests; +27 new)
- [x] Manually tested in all three popovers (Inspect / Draw / Flow) × both target states (running / dormant) = 6 cases
- [x] Edge cases: zero CLI sessions → primary falls back to "Send to New Session" and overflow menu also adapts. Outside-click dismisses. `max-height: 320px; overflow-y: auto` handles many sessions.

## Screenshots

Screenshot 2026-04-23 at 11 02 30 PM

## Notes

**Design decisions worth flagging:**
1. **Smart default = first running → first dormant → new** (stateless). No persisted "last target" — could be a follow-up.
2. **Inline popover, not modal**, for the overflow menu — models on `flowPickerMenu` in `pane.ts`. One click to pick a session vs. the 3-click modal flow.
3. **Session filter**: `!s.type || s.type === 'claude'`. Excludes browser-tab, mcp-inspector, diff-viewer, file-reader, remote-terminal — none are injectable PTY sessions.
4. **`injectPromptIntoRunningSession` respects `bracketedPasteMode`** on the xterm instance; falls through to raw payload if off. Always appends `\r` to submit.
5. **`setActiveSession(projectId, sessionId)` after delivery** so the UI auto-navigates to the target — no hunting.
6. **z-index 200** on `.send-menu-overlay` so the menu renders above `.browser-inspect-panel` (z-index 101).

**Not in this PR** (potential follow-ups):
- Remember last-used target per-project
- Provider icon in menu rows (claude / codex / gemini)
- Keyboard navigation inside the overflow menu (first item auto-focuses; arrow-key nav would complete it)
- Session ordering in the menu (currently project-array order)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/renderer/components/browser-tab/pane.ts [hunk 1]
```diff
@@ -218,8 +226,8 @@ export function createBrowserTabPane(sessionId: string, url?: string): void {
 
   const customBtn = document.createElement('button');
   customBtn.className = 'inspect-dropdown-btn';
-  customBtn.textContent = '\u25BC';
```
Comment (elirantutia) [inline src/renderer/components/browser-tab/pane.ts:221]: Please keep using this icon instead of `⋯`

**Comment by elirantutia** [comment]
Hey @MusharrafSakran thanks for the contribution.

Few points from my side before we can merge it - 
1. Don't auto choose session, the default is new session. Also use the `Send to AI` text for the default button.
2. Use the chevron icon as before instead of the three dots for more options.
3. I’m trying to understand - the dot next to the session already indicates its status, so why is there also text next to it?

Thank you very much!

File: src/renderer/components/browser-tab/pane.ts (hunk 1 above)
Comment (MusharrafSakran) [reply src/renderer/components/browser-tab/pane.ts:221]: Done in 143a716 ✅ — reverted to `▼`.

**Comment by MusharrafSakran** [comment]
Thanks for the review @elirantutia! Pushed `143a716` for **#2** (icon reverted to `▼`) and **#3** (dropped the italic status text — the dot already carries it, moved the full status into a hover `title`).

On **#1** — before I revert, wanted to share two things in case it reads differently now:

**1. The smart-default was imprecise. I just fixed it (`968fc96`).**

Previously it picked "first running session in array order" — which isn't what users expect. The new behavior walks the project's nav history backwards and picks **the most recently active CLI session** — i.e. the session that's currently highlighted (non-grayed-out) in the sidebar. So if you have Session 1, Session 2, Session 3 all open and you just clicked Session 2 before opening the browser tab, the primary button targets Session 2 — exactly what the sidebar says is "current". Falls back to first-running → first-dormant → new session only when there's no prior CLI session in the history.

**2. The concrete scenario it serves**

Mid-session debugging: I'm in Session 2 working on a feature → open the browser tab to see the rendered bug → click Inspect → Send. With always-new-session, I lose all the context Session 2 already has (files I've been editing, the plan it just made). With smart-default, the prompt lands right where the context lives. The `▼` overflow stays there for anyone who wants "new session" explicitly.

Two guards against "where did my prompt go?":
- Button label shows the target upfront (`Send to Session 2`) — no hidden destination
- `setActiveSession()` fires after delivery so the UI auto-navigates to the target

**If the behavior is fine but the dynamic label is the concern**, I can keep the smart-inject action but revert button text to static `Send to AI`. Or ship smart-default as an opt-in preference, off by default. Or full revert if you'd rather keep it simple.

Happy to do any of those — just wanted to surface the reasoning before ripping it out. 🙏

**Comment by elirantutia** [comment]
Thanks for the detailed reasoning! I hear you on the mid-session debugging case, but in most scenarios users are working on parallel UI changes across sessions, so defaulting to a new session is the safer assumption. Let's go with the full revert on #1 — default to new session, keep the `▼` overflow for explicit session targeting.

Appreciate the thorough writeup though 🙏

**Comment by MusharrafSakran** [comment]
Done @elirantutia — pushed `6121b00`:

- Primary `Send to AI` button now always creates a new session (smart-default removed entirely).
- Button label is static — no dynamic target-based text.
- ▼ overflow menu keeps the full session list for explicit targeting.
- Small QoL inside the ▼ menu: the CLI session that's currently active in the tab bar / swarm layout gets a subtle accent stripe + tint so it's easy to locate — doesn't change default behavior. The highlight also stays in sync if you switch tabs while the menu is open.

Net diff: +70 / −354.

File: src/renderer/state.ts [hunk 2]
```diff
@@ -91,6 +91,25 @@ class AppState {
...
+  /**
+   * Returns the most recently active CLI-backed session within the given project,
+   * walking the nav history backwards from the current position. Used by features
+   * that want to target "the session the user was just in" (e.g. browser-tab send).
+   * CLI-backed = no `type` set or `type === 'claude'`. Returns null if none found.
+   */
+  getLastActiveCliSessionInProject(projectId: string): SessionRecord | null {
+    const project = this.state.projects.find((p) => p.id === projectId);
+    if (!project) return null;
+    const isCli = (s: SessionRecord): boolean => !s.type || s.type === 'claude';
```
Comment (elirantutia) [inline src/renderer/state.ts]: why do you have this check `s.type === 'claude'`?

File: src/renderer/components/browser-tab/send-menu.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import { appState, type SessionRecord, type ProjectRecord } from '../../state.js';
+import { getTerminalInstance } from '../terminal-pane.js';
+import type { BrowserTabInstance } from './types.js';
+
+function isCliBacked(s: SessionRecord): boolean {
+  return !s.type || s.type === 'claude';
```
Comment (elirantutia) [inline src/renderer/components/browser-tab/send-menu.ts]: same here like the previous comment -  `s.type === 'claude'`?

**Comment by elirantutia** [comment]
Few notes to fix before I can merge it:

1. I don't understand what's the logic behind the selected session (I guess it shows the last session that it sent to before), I created a  fresh new session - 'Session 3' and it selected it.
image

2. When I see the options of the valid sessions, and then I close 1 of the session, and then I choose this session - I got a black screen. Make sure the session is still valid.
3. There is some legacy code with the `s.type === 'claude'` condition (what I wrote you in the code review) that I removed and refactored in the `main` branch, so please update your branch and consume the helper function `isCliSession` in `src/renderer/session-utils.ts`

**Comment by MusharrafSakran** [comment]
@elirantutia Pushed `ca2ef37` addressing all three points:

1. **Highlight removed.** No more selected-session logic in the menu — it just lists CLI sessions plain. Removes `getLastActiveCliSessionInProject` (now unused) and the related CSS.
2. **Black-screen on closed-session pick** — fixed two ways:
   - `deliver()` validates the picked session still exists in `project.sessions` before calling `setActiveSession` / `injectPromptIntoRunningSession` / `setPendingPrompt`. No more phantom-id active session.
   - The open menu subscribes to `session-removed`/`session-added` and re-renders in place, so closing a tab while the menu is up immediately drops that row.
   - Added a regression test for the stale-pick path.
3. **Adopted `isCliSession`** from `src/renderer/session-utils.ts` after merging `upstream/main`. Zero `s.type === 'claude'` checks left in the repo.

Net diff vs the previous push: +42 / −76 across 5 files. CI green on all three platforms.

**Comment by elirantutia** [comment]
Thanks @MusharrafSakran! Really appreciate you picking this up as your first contribution here, the back-and-forth on the design tradeoffs was genuinely great, and the final result is clean. Hope this is the first of many! Welcome to the community.

Merging now.
